test/Microsoft.ML.TestFramework/TestCommandBase.cs

```diff
@@ -2028,5 +2028,14 @@ public void CommandTrainingBinaryFieldAwareFactorizationMachineWithValidationAnd
             Assert.True(outputPath.CheckEqualityNormalized());
             Done();
         }
+
+        [Fact]
+        public void DataTypes()
+        {
+            string idvPath = GetDataPath("datatypes.idv");
+            OutputPath textOutputPath = CreateOutputPath("datatypes.txt");
+            TestCore("savedata", idvPath, "loader=binary", "saver=text", textOutputPath.Arg("dout"));
+            Assert.True(textOutputPath.CheckEquality());
+        }
```

Comment:
Right. We should firstly *not* be doing an `Assert.True` on check equality. The failure, if it happens, ought to be returned by the call to "Done" to allow a more comprehensive error. Also if there was a "soft failure" inside `TestCore`, we wouldn't be catching it *unless* we call `Done()`.

---
In reply to: [213145100](https://github.com/dotnet/machinelearning/pull/750#discussion_r213145100) [](ancestors = 213145100)

Reply:
If we are intending this to be a baseline test, it would be more typical to make this a "Check". The trouble with `Assert.True` is that it will fail on the first failure, rather than collecting *all* baseline differences. This makes it harder to resolve failing tests. But for now maybe we keep the code as is, but yes this should have been improved.

Not sure how this happened. This was done deliberately, which means someone actually made the test hardness worse on purpose when we moved the code.

But you still need the call to `Done()`...

---
In reply to: [213148985](https://github.com/dotnet/machinelearning/pull/750#discussion_r213148985) [](ancestors = 213148985,213145100)